### PR TITLE
Add support for building release tarballs

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2100/

File: Makefile [hunk 1]
```diff
@@ -122,11 +128,20 @@ endif
 gopkg:
 	@echo $(PROJECT)
 
+# Build a release tarball
+charmstore-$(VERSION).tar.xz: $(VERSIONDEPS)
+	mkdir -p charmstore-$(VERSION)/bin
+	GOBIN=$(CURDIR)/charmstore-$(VERSION)/bin go install $(INSTALL_FLAGS) -v $(PROJECT)/...
+	mv charmstore-$(VERSION)/bin/charmd charmstore-$(VERSION)/bin/charmstore
+	tar cfJv $@ charmstore-$(VERSION)
```
Comment (rogpeppe) [inline Makefile]: I had to look up the J flag (it's not even really documented, though I can guess). I wonder about using more usual primitives:

   tar c charmstore-$(VERSION) | xz > $@

?

**Review by rogpeppe — APPROVED** [review]
LGTM, but perhaps add charmstore-*.tar.xz to .gitignore too?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2101/

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Pull request merged**